product_contract/models/sale_order.py

```diff
@@ -9,19 +9,19 @@ class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
     @api.multi
-    def action_done(self):
-        ''' If we have a contract in the order, set it up '''
+    def action_confirm(self):
+        """ If we have a contract in the order, set it up """
         for rec in self:
-            if rec.state != 'done':
-                order_lines = rec.order_line.filtered(
-                    lambda r: r.product_id.is_contract==True
-                )
-                for line in order_lines:
-                    contract_tmpl = line.product_id.contract_template_id
-                    contract = self.env['account.analytic.account'].create({
-                        'name': '%s Contract' % rec.name,
-                        'partner_id': rec.partner_id.id,
-                        'contract_template_id': contract_tmpl.id,
-                    })
-                    contract.recurring_create_invoice()
-        super(SaleOrder, self).action_done()
+            order_lines = rec.order_line.filtered(
+                lambda r: r.product_id.is_contract==True
```

Comment:
No need to put `==True`. Every Python expression is evaluated as boolean when needed